internal/provider/string.go

```diff
@@ -219,7 +225,88 @@ func importStringFunc(sensitive bool) schema.StateContextFunc {
 		if sensitive {
 			d.SetId("none")
 		}
-		d.Set("result", val)
+		// common string passed to guess parameters based on the schema
+		// "password keepers={\"bla\": \"dibla\",\"key\":\"value\"},length=25,special=true,upper=true,lower=true,number=true,min_numeric=0,min_upper=0,min_lower=0,min_special=0,override_special=_%@"
+		// use a regex to fetch all settings
+		// Note: override_special should be the last specified key if specified
+		settings := map[string]interface{}{
+			// At some point we should use a default value here, why not 16 then.
+			"keepers":          nil,
+			"length":           16,
+			"special":          true,
+			"upper":            true,
+			"lower":            true,
+			"number":           true,
+			"min_numeric":      0,
+			"min_upper":        0,
+			"min_lower":        0,
+			"min_special":      0,
+			"override_special": "",
+		}
+
+		exploded := strings.Fields(val)
+
+		switch len(exploded) {
+		case 1:
+			// Keep default behavior but add default values
+			d.Set("result", val)
+			log.Println(fmt.Sprintf("[DEBUG] using default settings: %v", settings))
+			for key, value := range settings {
```

Comment:
It looks like this loop includes `length` from the map above, though that might not always match the length of `val`. Just curious, would it be reasonable to calculate the length of `val` and use that as the default here, rather than using 16 if a string is imported with no length specified?

Reply:
Hi there, I actually thought about it, but this causes issues with passwords that do not match the length specified in the resource definition.

eg. when you specify a length = 8 and importing 32 chars password, then the state of the resource will be 32 and not 8, causing a resource re recreation. 

Length in the import statement must match the length in the resource definition, but password length could be different.